Two issues in one patch:

We use shared libpci so we don't need to check for libraries it might be linking to, such as zlib.
Change mkdir_p with MKDIR_P for compability with recent autotools.

--- R-0.0.5/configure.ac
+++ R-0.0.5/configure.ac
@@ -14,6 +14,7 @@
 AC_PROG_INTLTOOL
 AC_GNU_SOURCE
 AC_HEADER_STDC
+AC_PROG_MKDIR_P
 
 AC_CHECK_FUNCS([vasprintf])
 
@@ -33,14 +34,6 @@
   echo "> Not using debug"
 fi
 
-# PCI library needs libz.
-
-AC_CHECK_HEADER([zlib.h],
-AC_CHECK_LIB(z,inflate,
-LIBS="$LIBS -lz"
-)
-)
-
 AC_CHECK_HEADER([pci/pci.h],
 AC_CHECK_LIB(pci,pci_alloc, 
 AC_DEFINE(HAVE_PCIUTILS,[1],[Use pci utils to discover devices])
